### PR TITLE
Feedback

### PR DESCRIPTION
**Pull request by underwaterhayk**
:wave:! GitHub Classroom created this pull request as a place for your teacher to leave feedback on your work. It will update automatically. **Don’t close or merge this pull request**, unless you’re instructed to do so by your teacher.

In this pull request, your teacher can leave comments and feedback on your code. Click the **Subscribe** button to be notified if that happens.

Click the **Files changed** or **Commits** tab to see all of the changes pushed to `master` since the assignment started. Your teacher can see this too.

<details>
<summary>
<strong>Notes for teachers</strong>
</summary>

Use this PR to leave feedback. Here are some tips:

- Click the **Files changed** tab to see all of the changes pushed to `master` since the assignment started. To leave comments on specific lines of code, put your cursor over a line of code and click the blue **+** (plus sign). To learn more about comments, read “[Commenting on a pull request](https://docs.github.com/en/github/collaborating-with-issues-and-pull-requests/commenting-on-a-pull-request)”.
- Click the **Commits** tab to see the commits pushed to `master`. Click a commit to see specific changes.
- If you turned on autograding, then click the **Checks** tab to see the results.
- This page is an overview. It shows commits, line comments, and general comments. You can leave a general comment below.

For more information about this pull request, read “[Leaving assignment feedback in GitHub](/help/leaving-feedback-in-github)”.
</details>

Subscribed: @RudolfStepanian

### CODE REVIEW

**Comment by azakhary** [comment]
Nice showcase of generics, but here are some issues with this PR

- No proper gitignore file, changes contain IDE configuration files that should have been excluded
- This is overall bad approach, since in a nutshell this could have been many-to-many material exchange. As described, payload is many items, and cost is many items. In theory I can have a product which allows me to use A & B to purchase C & D in one transaction. Current structure is limited, only supports one to one transactions(A for B) and furthermore for no apparent reason separates currency from purchasable goods, while they could have been the same to allow for flexibility. (You never know how game will change in this regard, so need to have flexible code)
- Ensuring the transaction is a job of shop, shop holds the products and it should act using them on a player interface. Currently your player is doing the transaction. This is similar if you go to the mall and instead of cashier do the all job yourself. It does not make sense in real world, thus it does not make sense in code (buy method should be part of shop, and get player as param to act upon it)

File: src/com/rockbite/bootcamp/store/IInventory.java [hunk 1]
```diff
@@ -0,0 +1,12 @@
+package com.rockbite.bootcamp.store;
+
+import com.rockbite.bootcamp.store.collections.Resources.Item;
+
+public interface IInventory {
+
+    boolean hasItem(Item item, int count);
+
+    void spendItem(Item item, int count);
```
Comment (Tom-Ski) [inline src/com/rockbite/bootcamp/store/IInventory.java:9]: spendItem, suggests that you are paying for something with an item. What are you purchasing in exchange for this?

File: src/com/rockbite/bootcamp/store/Pool/Pool.java [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+        usedObjects.add(object);
+        return object;
+    }
+
+    public T obtain(T object){
+        usedObjects.add(object);
+        return object;
+    }
+
+    public void free(T object){
```
Comment (Tom-Ski) [inline src/com/rockbite/bootcamp/store/Pool/Pool.java:28]: Inconsistent method formatting

File: src/com/rockbite/bootcamp/store/Product.java [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+    private Resources price = new Resources();
+
+    public Product(String name) {
+        this.name = name;
+    }
+
+    public void addPrice(Item item, Integer count){
+        price.addItem(item,count);
+    }
+    public void addPayload(Item item, Integer count){
```
Comment (Tom-Ski) [inline src/com/rockbite/bootcamp/store/Product.java:20]: Inconsistent spacing around method params and brackets

File: src/com/rockbite/bootcamp/store/Shop/IShop.java [hunk 4]
```diff
@@ -0,0 +1,10 @@
+package com.rockbite.bootcamp.store.Shop;
+
+import com.rockbite.bootcamp.store.IInventory;
+import com.rockbite.bootcamp.store.Product;
+
+public interface IShop {
+    void getProduct(Product it, int count);
```
Comment (Tom-Ski) [inline src/com/rockbite/bootcamp/store/Shop/IShop.java:7]: Why do we need a count here in getProduct? getProduct implies we are just returning a product, but this doesn't return anything, and it has parameters that are undefined.

File: src/com/rockbite/bootcamp/store/Shop/IShop.java [hunk 5]
```diff
@@ -0,0 +1,10 @@
...
+    boolean canAfford(IInventory user, Product it);
+    void transaction(IInventory user, Product it);
```
Comment (Tom-Ski) [inline src/com/rockbite/bootcamp/store/Shop/IShop.java:9]: This method does not give any information about what is transacting.

File: src/com/rockbite/bootcamp/store/Shop/Shop.java [hunk 6]
```diff
@@ -0,0 +1,154 @@
...
+    }
+
+    @Override
+    public void getProduct(Product product, int count) {
+
+    }
+
+    @Override
+    public boolean canAfford(IInventory user, Product product) {
+        if (productList.getCollection().get(product)!=null){
```
Comment (Tom-Ski) [inline src/com/rockbite/bootcamp/store/Shop/Shop.java:83]: Inconsistent spacing around control structures. 

if (){ in the first line, 
for (...) { in the second line

**Pull request merged**